plugin/src/main/java/org/opensearch/ml/action/connector/GetConnectorTransportAction.java

```diff
@@ -49,25 +49,36 @@ public class GetConnectorTransportAction extends HandledTransportAction<ActionRe
 
     ConnectorAccessControlHelper connectorAccessControlHelper;
 
+    private final MLFeatureEnabledSetting mlFeatureEnabledSetting;
+
+    private TenantAwareHelper tenantAwareHelper;
+
     @Inject
     public GetConnectorTransportAction(
         TransportService transportService,
         ActionFilters actionFilters,
         Client client,
         SdkClient sdkClient,
-        ConnectorAccessControlHelper connectorAccessControlHelper
+        ConnectorAccessControlHelper connectorAccessControlHelper,
+        MLFeatureEnabledSetting mlFeatureEnabledSetting
+
     ) {
         super(MLConnectorGetAction.NAME, transportService, actionFilters, MLConnectorGetRequest::new);
         this.client = client;
         this.sdkClient = sdkClient;
         this.connectorAccessControlHelper = connectorAccessControlHelper;
+        this.mlFeatureEnabledSetting = mlFeatureEnabledSetting;
+        this.tenantAwareHelper = new TenantAwareHelper(mlFeatureEnabledSetting);
     }
 
     @Override
     protected void doExecute(Task task, ActionRequest request, ActionListener<MLConnectorGetResponse> actionListener) {
         MLConnectorGetRequest mlConnectorGetRequest = MLConnectorGetRequest.fromActionRequest(request);
         String connectorId = mlConnectorGetRequest.getConnectorId();
         String tenantId = mlConnectorGetRequest.getTenantId();
+        tenantAwareHelper.setTenantId(tenantId);
```

Comment:
Is this thread safe?

Reply:
I've decided to pass that as a parameter, so it won't be here.

Reply:
removing this setter altogether.